application-urlshortener-ui/src/main/resources/URLShortener/WebHome.xml

```diff
@@ -36,5 +36,308 @@
+        <editor>PureText</editor>
+        <name>code</name>
+        <number>2</number>
+        <prettyName>Code</prettyName>
+        <restricted>0</restricted>
+        <rows>20</rows>
+        <size>50</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </code>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>require(['jquery', 'urlshortener-localization'], function ($, l10n) {
+  function regenerateURL(element) {
+    element = element.currentTarget;
+    const oldPageID = element.getAttribute('data-old-page-id');
+    const pageRef = element.getAttribute('data-page-ref');
+    const postURL = '/xwiki/rest/p/regenerate?' + new URLSearchParams({ oldPageID: oldPageID, currentDocRef: pageRef });
+    let notification = new XWiki.widgets.Notification(l10n.regenerateInProgress, 'inprogress');
+    $.post(postURL, {}).done(function(response) {
+      notification.replace(new XWiki.widgets.Notification(l10n.regenerateDone, 'done'));
+      $(element).prop('disabled', 'true');
+    }).fail(function(error) {
+      let failMessage = (error.status == 401) ? l10n.regenerateFailUnauthorized : l10n.regenerateFail;
+      notification.replace(new XWiki.widgets.Notification(failMessage, 'error'));
+    });
+  };
+  $('.urlshortener-regenerate-btn').click(regenerateURL);
+});</code>
+    </property>
+    <property>
+      <name>Regenerate URL</name>
+    </property>
+    <property>
+      <parse>0</parse>
+    </property>
+    <property>
+      <use>currentPage</use>
+    </property>
+  </object>
+  <object>
```

Comment:
You don't need a new JS object only for this code, as it is linked to what you have above. So only one JS object is enough